src/nunit.analyzers/UseSpecificConstraint/UseSpecificConstraintAnalyzer.cs

```diff
@@ -45,11 +45,12 @@ invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpr
 
                 if (argument is LiteralExpressionSyntax literalExpression)
                 {
-                    constraint = literalExpression.Kind() switch
+                    constraint = (literalExpression.Kind(), literalExpression.Token.ValueText) switch
                     {
-                        SyntaxKind.NullLiteralExpression => NUnitFrameworkConstants.NameOfIsNull,
-                        SyntaxKind.FalseLiteralExpression => NUnitFrameworkConstants.NameOfIsFalse,
-                        SyntaxKind.TrueLiteralExpression => NUnitFrameworkConstants.NameOfIsTrue,
+                        (SyntaxKind.NullLiteralExpression, _) => NUnitFrameworkConstants.NameOfIsNull,
+                        (SyntaxKind.FalseLiteralExpression, _) => NUnitFrameworkConstants.NameOfIsFalse,
+                        (SyntaxKind.TrueLiteralExpression, _) => NUnitFrameworkConstants.NameOfIsTrue,
+                        (SyntaxKind.NumericLiteralExpression, "0") => NUnitFrameworkConstants.NameOfIsZero,
```

Comment:
Suggest not to switch on both, but treat the numeric part different. Which allow checking some more 0 values.